flytekit/tools/script_mode.py

```diff
@@ -46,6 +50,8 @@ def compress_scripts(source_path: str, destination: str, module_name: str):
 
         visited: typing.List[str] = []
         copy_module_to_destination(source_path, destination_path, module_name, visited)
+        add_imported_modules_from_source(source_path, destination_path, list(sys.modules.values()))
```

Comment:
If we remove `copy_module_to_destination`, does it work as well? If so, I think we can remove that.

For more context:
`copy_module_to_destination` will only copy the module to remote if and only if that module has a flyte task/workflow.


in this case, pyflyte run --remote uploads `foo_wf` and `main`
```python
└── workflows
    ├── __init__.py
    ├── another_module
    │   └── foo_wf.py
    ├── main.py  # it import foo_wf
```

I did this because I didn't know how to do `add_imported_modules_from_source` at that time. 

Now you have added it, I think we should be able to remove `copy_module_to_destination`

Reply:
+1, remove this one line and merge.  tested this against dev but feel free to again.  the only difference i saw is that with the copy_module... call, the main script itself is already in the destination folder, and because it exists it's skipped when add_imported... sees it.  Without it, the main file itself is copied by add_imported...